File: 프로그래머스/0/181851. 전국 대회 선발 고사/전국 대회 선발 고사.js
function solution(rank, attendance) {
  let arr = [];
  attendance.map((x, i) => {
    if(x){
      arr.push(rank[i])
    }
  })
  const rankArr = arr.sort((a, b) => a - b).map(x => rank.indexOf(x))
  return (10000 * rankArr[0]) + (100 * rankArr[1]) + rankArr[2]
}

File: 프로그래머스/0/181851. 전국 대회 선발 고사/전국 대회 선발 고사.java
import java.util.*;

class Solution {
    public int solution(int[] rank, boolean[] attendance) {
        List<int[]> students = new ArrayList<>();
        for (int i = 0; i < rank.length; i++) {
            if (attendance[i]) {
                students.add(new int[]{rank[i], i});
            }
        }

        students.sort(Comparator.comparingInt(a -> a[0]));
        
        int a = students.get(0)[1];
        int b = students.get(1)[1];
        int c = students.get(2)[1];

        return 10000 * a + 100 * b + c;
    }
}